商品削除機能

miracletea opened the pull request:
# What
商品削除機能の作成

# Why
商品削除機能の実装のため

【Gyazo】
ログイン状態の出品者のみ、詳細ページの削除ボタンを押すと、出品した商品を削除できる動画
https://gyazo.com/11571cf7048b2c3acc3c4f359d6846c1

miracletea committed:
destroy機能の作成

miracletea committed:
商品出品の制限見直し

div-reviewer reviewed (commented):
miracletea様、学習お疲れさまです。
商品削除の初回レビュー依頼ありがとうございました。

完成度高く実装されているかと思われます。今までの学習がバッチリ成果として出ていますね。大変素晴らしいです。
こちら`LGTM`とさせて頂きます。実装お疲れさまでした。

**●任意修正について●**
タイトルにあります通り、修正は任意です。
よりよいコードを書くための知識としてご確認いただけますと幸いです。

修正して頂ける場合は`再度レビュー依頼は出さなくて大丈夫です`ので、修正後、コミットとプッシュして頂けましたら、
マージ後、ローカルのmainブランチに忘れずにプルし、次の実装に取り組んで頂ければと思います。

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
今後の実装も応援しております！どうぞよろしくおねがいします。**

miracletea committed:
【任意修正】set_itemのmove_to_pageを削除

app/controllers/items_controller.rb
Comment on lines +2 to 4
  before_action :set_item, only: [:show, :edit, :update, :move_to_page, :destroy]
  before_action :move_to_page, except:[:index, :show, :new, :create]
  before_action :authenticate_user!, except:[:index, :show]
@div-erika div-erika 8 minutes ago
こちらは【任意修正】となります。
よりよいコードを書くための知識としてご確認いただけますと幸いです。

  before_action :set_item, only: [:show, :edit, :update, :move_to_page, :destroy]
before_action :set_itemからmove_to_pageは削除していただいても大丈夫です。

  before_action :set_item, only: [:show, :edit, :update, :move_to_page, :destroy]
  before_action :move_to_page, except:[:index, :show, :new, :create]
  before_action :authenticate_user!, except:[:index, :show]
理由：
上記の記述によって、set_itemの次にmove_to_pageが動くようになっています。

miracletea committed:
【任意修正】if文条件の見直しuser_signed_in?

app/controllers/items_controller.rb
Comment on lines +39 to +44
    if user_signed_in?
      if current_user.id == @item.user.id
        @item.destroy
        redirect_to items_path
      end
    end
@div-erika div-erika 9 minutes ago
こちらは【任意修正】となります。
よりよいコードを書くための知識としてご確認いただけますと幸いです。

if user_signed_in?
上記は削除していただいても大丈夫です。before_action :authenticate_user!, except:[:index, :show]によってログインしているユーザーしかdestroyに入らないためです。
